### PR TITLE
Update API for node2vec and biased random walks

### LINKED ISSUE #2558: MNMG Random Walks - Python
Modify the python Random Walks calls to use the new PLC version of Random Walks. Add MNMG support.

API definition in https://github.com/rapidsai/cugraph/issues/2552 blocks starting this work
Implementation in https://github.com/rapidsai/cugraph/issues/2555 blocks completing this work

### LINKED ISSUE #2559: MNMG Random Walks - Python Unit Tests
Implement python tests for MG Random Walks.  Should include tests where random walks cross GPU boundaries.  Should include tests for uniform, biased and node2vec walks are computed.

### LINKED ISSUE #2560: MNMG Random Walks - Remove old Python
Remove old python implementation of Random Walks.

Blocked by https://github.com/rapidsai/cugraph/issues/2558 for implementation

### LINKED ISSUE #2561: MNMG Random Walks - Remove legacy C/C++ code
Remove legacy C/C++ code for Random walks.  This should include the legacy C API implementation as well as the legacy C++ implementation.

Blocked by https://github.com/rapidsai/cugraph/issues/2560

### LINKED ISSUE #2853: [FEA]: Add python implementation of MG node2vec
### Is this a new feature, an improvement, or a change to existing functionality?

New Feature

### How would you describe the priority of this feature request

High

### Please provide a clear description of problem this feature solves

There is currently no python implementation of MG `node2vec` even though it is supported by the CAPI. This feature
will enable MG node2vec runs which is currently only supported on  SG

### Describe your ideal solution

1. Add a python implementation of MG `node2vec` following the PLC path
2. Add docstrings
3. Add coverage/tests

### LINKED ISSUE #3277: Algorithm Migration - Random Walks
```[tasklist]
### Tasks
- [ ] #2552
- [ ] #2553 
- [ ] #2554 
- [ ] #2555 
- [ ] #2556 
- [ ] #2557 
- [ ] #2558 
- [ ] #2559 
- [ ] #2560 
- [ ] #2561 
```

### LINKED ISSUE #4843: [QST]: Does ECG support multi-GPU?
### What is your question?

Does ECG support multi-GPU? Can not find in documentation, but find multi-gpu setting in source code.

### Code of Conduct

- [x] I agree to follow cuGraph's Code of Conduct
- [x] I have searched the [open issues](https://github.com/rapidsai/cugraph/issues) and have found no duplicates for this question

### PR DESCRIPTION
**Pull request by jnke2016**
This PR

1. Add MG support for node2vec random walks on the C, PLC and Python API
2. Fix renumbering bug in the CAPI for node2vec and biased random walks
3. Enable and fix bug in the MG C tests
4. Deprecate legacy node2vec and uniform random walks
5. Add SG and MG support for Biased random walks
6. Add multi GPU support for ECG in the python API

closes #3277 
closes #2558
closes #2559 
closes #2560 
closes #2561 
closes #2853 
closes #4843

### CODE REVIEW

File: python/cugraph/cugraph/sampling/random_walks.py [hunk 1]
```diff
@@ -126,13 +128,39 @@ def random_walks(
...
+    # Context for legacy_result_type: The initial implementation of random_walks
+    # returned results where the vertex and weight path are proportional to the
+    # number of vertices instead of the number of edges hence the flag
+    # 'legacy_result_type' was created. This flag should be removed in favor of
+    # returning results paths proprtional to the number of edges. Furthermore,
+    # Coalesced path results should also be removed in favor of always returning
+    # padded results. The flags 'legacy_result_type' and 'use_padding" should be
+    # removed. 
+
+    if legacy_result_type or use_padding is False:
```
Comment (ChuckHastings) [inline python/cugraph/cugraph/sampling/random_walks.py:148]: Note... it seems like the old mechanism (coalesced format) is not particularly helpful other than a modest savings in memory on a small result type anyway.

We don't intend to support that in the C++ code any longer.  If we need to continue to support this old mechanism while it is deprecated before we can delete it I would suggest implementing something to convert the format within the python layer.

Comment (jnke2016) [reply python/cugraph/cugraph/sampling/random_walks.py:156]: We still have the old mechanism around and I deprecated the coalesced path in the python API. Since there is no real benefit of using the coalesced path, I suggest to not support in the new API. @rlratzel any thoughts?

**Review by ChuckHastings — APPROVED** [review]

**Review by rlratzel — APPROVED** [review]
Reviewed the Python parts. I had just a few optional suggestions.
